junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/ExtensionContext.java

```diff
@@ -401,6 +402,17 @@ default void publishReportEntry(String value) {
 	@API(status = STABLE, since = "5.11")
 	ExecutableInvoker getExecutableInvoker();
 
+	/**
+	 * Returns a list of registered extension at this context of the passed {@code extensionType}.
+	 *
+	 * @param <E> the extension type
+	 * @param extensionType the extension type
+	 * @return the list of extensions
+	 * @since 5.12
+	 */
+	@API(status = EXPERIMENTAL, since = "5.12")
+	<E extends Extension> List<E> getExtensions(Class<E> extensionType);
+
```

Comment:
I discussed the addition of this method to the public API with the rest of the team today. There was some concern this could be (mis)used to interact or mutate other extensions. On the other hand, it could be useful for use cases such as #1945. Ultimately, we decided to not expose it for now since doing so requires more thought to be put into possible implications. Therefore, please introduce a `org.junit.jupiter.engine.extension.ExtensionContextInternal` interface that extends `ExtensionContext` adding this method, have `AbstractExtensionContext` implement it, and downcast in `TemporaryDirectory` before passing it to `PreInterruptCallbackInvocationFactory.create`.

Reply:
Got it. I will have a look at it tomorrow.